### PR TITLE
Work on consistency of `Operator`s

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
In the context of preparing a tool for users that allows to analyze operations for consistency, correctness and completeness, some minor inconsistencies were found in PennyLane's operations and observables.

**Description of the Change:**
- Very minor speedup of string comparisons in `PauliError`
- Formatting/Clean up of some error messages
- Addition of `num_params=0` to `WireCut`
- ~~Change `SnapShot.wires` from `[]` to `qml.operation.AllWires`~~
- Remove `MultiControlledX` from parametrized tests in `test_parametric_ops.py`
- Remove `qml.SWAP.basis="X"`

**Benefits:**
Code quality and consistency.

**Possible Drawbacks:**
N/A

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -297,6 +297,14 @@
 
 <h3>Improvements</h3>
 
+* Some error messages have been improved and operation properties made
+  more consistent.
+  [(#2287)](https://github.com/PennyLaneAI/pennylane/pull/2287/)
+
+* Most compilation transforms, and relevant subroutines, have been updated to
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply doc/releases/changelog-dev.md:304]: Oh dear, yes, I overlooked this one when merging.

File: pennylane/ops/channel.py [hunk 2]
```diff
@@ -533,12 +533,12 @@ def __init__(self, operators, p, wires=None, do_queue=True, id=None):
         super().__init__(operators, p, wires=wires, do_queue=do_queue, id=id)
 
         # check if the specified operators are legal
-        if not all(c in "XYZ" for c in operators):
+        if not set(operators).issubset({"X", "Y", "Z"}):
```
Comment (albi3ro) [inline pennylane/ops/channel.py:536]: 🚀 👍

**Review by albi3ro — APPROVED** [review]
Thanks for these updates! A lot of this small stuff can be easily overlooked, so it's great to catch them.

**Comment by dwierichs** [comment]
I suppose this would be ready to merge once CodeFactor converges on an opinion :)

**Pull request merged**